File: Java-Src/NodeTronModel.java
import java.util.*;

public class NodeTronModel extends Network
{
    public ArrayList<ArrayList<Double>> nodeCases = new ArrayList<ArrayList<Double>>();

    public NodeTronModel(double learningRate, ArrayList<Layer> layers)
    {
        super(learningRate, layers);
    }
    
    @Override
    public void buildNetwork()
    {
        System.out.println("Starting to build NodeTronModel");
        Node mainNode = new Node(2, 1, "mainNode", true);
        Layer hiddenLayer = new Layer(this.getLearningRate(), "hiddenlayer1");
        ArrayList<State> inputLayer = new ArrayList<State>();
        inputLayer.add(new State());
        inputLayer.add(new State());
        ArrayList<State> outputLayer = new ArrayList<State>();
        outputLayer.add(new State());

        for (State state : inputLayer)
        {
            this.connectStateToNode(state, mainNode);
        }
        for (State state : outputLayer)
        {
            this.connectNodeToState(mainNode, state);
        }

        hiddenLayer.addNode(mainNode);

        this.getLayers().add(hiddenLayer);

        this.setOutputLayer(outputLayer);
        this.setInputLayer(inputLayer);

        System.out.println("Finished building NodeTronModel");
    }

    @Override
    public void trainNetwork(int iterations, ArrayList<ArrayList<Double>> cases)
    {
        Random rand = new Random();
        while (iterations > 0)
        {
            ArrayList<Double> inst = cases.get(rand.nextInt(cases.size()));
            ArrayList<Double> forwardList = new ArrayList<Double>();
            forwardList.add(inst.get(0));
            forwardList.add(inst.get(1));
            
            ArrayList<Double> backwardList = new ArrayList<Double>();
            backwardList.add(inst.get(2));

            this.runForward(forwardList);
            this.runBackward(backwardList);
            iterations--;
        }
    }
}

File: Csharp-Src/Csharp-Src/NodeTronModel.cs
using System;
using System.Collections.Generic;

namespace Csharp_Src
{
    public class NodeTronModel : Network
    {
        public NodeTronModel() : base()
        {
        }

        public override void BuildNetwork()
        {
            Node mainNode = new Node(2, 1, "mainNode");
            Layer hiddenLayer = new Layer(this.LearningRate);
            List<State> inputLayer = new List<State>() { new State(), new State() };
            List<State> outputLayer = new List<State>() { new State() };

            foreach (var state in inputLayer)
                this.ConnectStateToNode(state, mainNode);
            foreach (var state in outputLayer)
                this.ConnectNodeToState(mainNode, state);

            hiddenLayer.LayerNodes.Add(mainNode);

            this.Layers.Add(hiddenLayer);
            this.OutputLayer = outputLayer;
            this.InputLayer = inputLayer;
        }

        public override void TrainNetwork(int iterations, List<List<double>> cases)
        {
            Random rand = new Random();
            while (iterations > 0)
            {
                List<double> inst = cases[rand.Next(0, cases.Count)];
                this.RunForward(new List<double> { inst[0], inst[1] });
                this.RunBackward(new List<double> { inst[2] });
                iterations--;
            }
        }
    }
}
